pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -150,6 +155,14 @@ def __str__(self):
         items += "\ngate_types:\n{" + gate_type_str + "}"
         return items
 
+    def __repr__(self):
+        """Compact string representation of the Resources object"""
+        return {
+            "gate_types": self.gate_types,
+            "num_gates": self.num_gates,
+            "num_wires": self.num_wires,
+        }.__repr__()
+
```

Comment:
Why did we need this?

Reply:
When I added `__eq__` I got a bunch of errors in my tests, but they were difficult to read without a `__repr__` method defined.

Reply:
Then we need a simple test for this method too